okio-files/src/commonTest/kotlin/okio/files/PathTest.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio.files
+
+import okio.Path.Companion.toPath
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class PathTest {
+  @Test
+  fun `basic paths`() {
+    assertEquals("gradlew", "./gradlew".toPath().toString())
+    assertEquals(".gitignore", "./.gitignore".toPath().toString())
+    assertEquals("/home/jesse", "/home/jesse".toPath().toString())
+    assertEquals("/home/jake", "/home/jesse/../jake".toPath().toString())
+    assertEquals("../../etc/passwd", "../../etc/passwd".toPath().toString())
+  }
+
+  @Test
+  fun `parent of absolute path`() {
+    assertEquals("/home", "/home/jesse".toPath().parent?.toString())
+    assertEquals("/", "/home".toPath().parent?.toString())
+    assertNull("/".toPath().parent)
+  }
+
+  @Test
+  fun `parent of relative path`() {
+    assertEquals("src/main", "src/main/java".toPath().parent?.toString())
+    assertEquals("src", "src/main".toPath().parent?.toString())
+    assertEquals("", "src".toPath().parent?.toString())
+    assertNull("".toPath().parent)
+  }
+
+  @Test
+  fun `parent of relative path with single traversal`() {
+    assertEquals("../src/main", "../src/main/java".toPath().parent?.toString())
+    assertEquals("../src", "../src/main".toPath().parent?.toString())
+    assertEquals("..", "../src".toPath().parent?.toString())
+    assertNull("..".toPath().parent)
+  }
+
+  @Test
+  fun `parent of relative path with multiple traversal`() {
+    assertEquals("../../src/main", "../../src/main/java".toPath().parent?.toString())
+    assertEquals("../../src", "../../src/main".toPath().parent?.toString())
+    assertEquals("../..", "../../src".toPath().parent?.toString())
+    assertNull("../..".toPath().parent)
+  }
+
+  @Test
+  fun `absolute path traversal with div operator`() {
+    val root = "/".toPath()
+    assertEquals("/home".toPath(), root / "home")
+    assertEquals("/home/jesse".toPath(), root / "home" / "jesse")
+    assertEquals("/home".toPath(), root / "home" / "jesse" / "..")
+    assertEquals("/home/jake".toPath(), root / "home" / "jesse" / ".." / "jake")
```

Comment:
do I go to programming prison for this?

Reply:
yes :wink:

Reply:
🚔 🚔 🚔

Reply:
Just happened to stumble across this. I've done this in several projects that run on both Linux and Windows. It's amazing how humans have the ability to both love and detest the same thing at the same time :smile: